client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/MlClientDocumentationIT.java

```diff
@@ -749,6 +750,10 @@ public void testUpdateDatafeed() throws Exception {
             DatafeedConfig updatedDatafeed = response.getResponse(); // <1>
             // end::update-datafeed-response
             assertThat(updatedDatafeed.getId(), equalTo(datafeedId));
+        } catch (WarningFailureException e) {
```

Comment:
I think we should remove the `setJobId` method call earlier (or at least set the JobId to null so that it does not throw the exception), annotate `setJobId` as deprecated in the HLRC and state as much in the docs.

Additionally, it would be good to add an integration test for the deprecation exception separately from this test method. The integration test should probably live in `MachineLearningIT`